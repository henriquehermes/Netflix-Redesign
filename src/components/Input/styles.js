import styled from 'styled-components/native';

export const CustomInput = styled.TextInput`
  border: 2px solid ${props => props.theme.inputBackground};
  background-color: ${props => props.theme.inputBackground};
  border-radius: 4px;
  color: rgba(255, 255, 255, 0.87);
  font-size: 14px;
  width: 100%;
  max-width: 300px;
  padding: 5px 10px;
  margin-bottom: 10px;
`;
