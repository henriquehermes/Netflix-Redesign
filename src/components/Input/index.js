import React from 'react';

import { CustomInput } from './styles';

export default function Input({ placeHolder, type, secureTextEntry }) {
  return (
    <CustomInput
      placeholder={placeHolder}
      placeholderTextColor="rgba(255,255,255,0.2)"
      type={type}
      secureTextEntry={secureTextEntry}
    />
  );
}
