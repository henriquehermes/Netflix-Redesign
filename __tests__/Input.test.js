/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../src/components/Input';

it('renders correctly', () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});
