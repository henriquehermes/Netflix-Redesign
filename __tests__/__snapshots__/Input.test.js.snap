// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  placeholderTextColor="rgba(255,255,255,0.2)"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#1c1c1c",
        "borderColor": "#1c1c1c",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 2,
        "color": "rgba(255, 255, 255, 0.87)",
        "fontSize": 14,
        "marginBottom": 10,
        "maxWidth": 300,
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
        "width": "100%",
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
